rustls-platform-verifier/src/verification/others.rs

```diff
@@ -115,7 +115,7 @@ impl Verifier {
             );
         }
 
-        #[cfg(all(target_os = "linux", not(target_arch = "wasm32")))]
+        #[cfg(all(unix, not(target_os = "android"), not(target_os = "macos"), not(target_os = "ios"), not(target_os = "tvos")))]
```

Comment:
Is the removal of `not(target_arch = "wasm32")` intentional? 

I _think_ that `wasm32-unknown-emscripten` (as one example) is `arch == wasm32` and `target-family == [unix, wasm]` and would be included here even though there's no native certs to load for that target. Or am I overlooking something?

Reply:
@cpu To be frank I'm not sure, as the original request was:

> I guess this should be equivalent to the cfg expression that adds the dependency on rustls-native-certs in the first place.

Please feel free to adjust the guard as you see fit.

Reply:
I think probably both the `Cargo.toml` dependency gating for `rustls-native-certs` and this should have a `not(target_arch = "wasm32")` and it was probably an oversight :thinking: 

Does that make sense to you @ctz?

Reply:
Yes, that makes sense to me. There's no point in calling or building `rustls-native-certs` for wasm. (And if someone wants that to work for wasi we can deal with that when they ask.)

Reply:
> Please feel free to adjust the guard as you see fit.

I'll push a revision in a moment.